Source/JavaScriptCore/debugger/DebuggerParseData.cpp

```diff
@@ -30,6 +30,41 @@
 
 namespace JSC {
 
+void DebuggerPausePositions::forEachBreakpointLocation(int startLine, int startColumn, int endLine, int endColumn, Function<void(const JSTextPosition&)>&& callback)
+{
+    auto isBeforeStart = [&] (int line, int column) {
+        return line < startLine || (line == startLine && column < startColumn);
+    };
+
+    auto isAfterEnd = [&] (int line, int column) {
+        return (line == endLine && column >= endColumn) || line > endLine;
+    };
+
+    Vector<JSTextPosition> uniquePositions;
+    for (const auto& position : m_positions) {
+        auto line = position.position.line;
+        auto column = position.position.column();
+
+        if (isBeforeStart(line, column))
```

Comment:
I think this is probably fine right now, but if it is easy you may want to binary search to get to the start position.

I'm imagining a very large JavaScript file and someone is trying to set a breakpoint near the very end. I bet it is fast enough now, but I bet it would be easy to make this consistently faster.

Reply:
This is assuming that `m_positions` is sorted. Since code below appears to sort the results, maybe that isn't the case right now? But the `break` for `isAfterEnd` makes me think they were sorted.

Reply:
i think i may actually be able to refactor this a bit to avoid the repeated `std::lower_bound`

yes this should be sorted after creation in `gatherDebuggerParseData`